packages/metrics-prometheus/src/index.ts

```diff
@@ -330,6 +345,122 @@ class PrometheusMetrics implements Metrics {
       return counterGroup
     }
   }
+
+  registerHistogram (name: string, opts: PrometheusCalculatedHistogramOptions): void
+  registerHistogram (name: string, opts?: HistogramOptions): Histogram
+  registerHistogram (name: string, opts: any = {}): any {
+    if (name == null ?? name.trim() === '') {
+      throw new Error('Histogram name is required')
+    }
+
+    let metric = metrics.get(name)
+
+    if (metrics.has(name)) {
+      this.log('Reuse existing histogram', name)
+
+      if (opts.calculate != null) {
+        metric.addCalculator(opts.calculate)
+      }
+
+      return metrics.get(name)
+    }
+
+    this.log('Register histogram', name)
+    metric = new PrometheusHistogram(name, { registry: this.registry, ...opts })
+
+    metrics.set(name, metric)
+
+    if (opts.calculate == null) {
+      return metric
+    }
+  }
+
+  registerHistogramGroup (name: string, opts: PrometheusCalculatedHistogramOptions<Record<string, number>>): void
+  registerHistogramGroup (name: string, opts?: HistogramOptions): HistogramGroup
+  registerHistogramGroup (name: string, opts: any = {}): any {
+    if (name == null ?? name.trim() === '') {
+      throw new Error('Histogram group name is required')
+    }
+
+    let metricGroup = metrics.get(name)
+
+    if (metricGroup != null) {
+      this.log('Reuse existing histogram group', name)
```

Comment:
for consistency with other logs:

```suggestion
      this.log('reuse existing histogram group', name)
```